libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -908,6 +908,44 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 
 		break;
 	}
+	case FunctionType::Kind::Revert:
+	{
+		solAssert(arguments.size() == parameterTypes.size(), "");
+		if (arguments.empty())
+			m_code << "revert(0, 0)\n";
+		else
+			solUnimplementedAssert(false, "");
+
+		break;
+	}
+	case FunctionType::Kind::Send:
+	case FunctionType::Kind::Transfer:
+	{
+		solAssert(arguments.size() == 1 && parameterTypes.size() == 1, "");
+		string address{IRVariable(_functionCall.expression()).part("address").name()};
+		string value{expressionAsType(*arguments[0], *(parameterTypes[0]))};
+		Whiskers templ(R"(
+		let <gas> := 0
+		if iszero(<value>) { <gas> := <callStipend> }
+		let <success> := call(<gas>, <address>, <value>, 0, 0, 0, 0)
+		<?isTransfer>
+			if iszero(<success>) { <forwardingRevert>() }
+		</isTransfer>
+		)");
+		templ("gas", m_context.newYulVariable() + "_gas");
```

Comment:
@chriseth i'm wondering whether the resulting yul code would better readable if the "parameter name" would be always automatically prepended like here.

Reply:
I think we should provide a prefix instead. Note that this code will be something like `_34_gas`, while `_gas_34` will be shortened to `_gas` if there are no conflicts.

At the current point in time, this might be too dangerous - we already have so many prefixes we hope will not crash. Maybe better to just use `m_context.newYulVariable()` for now, without any prefix or suffix.